APE26.rst

```diff
@@ -0,0 +1,325 @@
+but not identical APIs:
+
+.. code-block:: python
+
+    c1 = SkyCoord(1., 2., frame="icrs", units="deg")
+    c2 = ICRS(1. * u.deg, 2. * u.deg)
+
+We suggest the situation should instead be analogous to what is the 
+case for units, which know how to transform from one to another, but 
+are not concerned with how the values are stored (that belongs to 
+``Quantity``). Translating to coordinates, units are like 
+``CoordinateFrame``, the values are the coordinate data 
+(``Representations``), and ``Quantity`` is like ``SkyCoord``.
+
+Having both the frame classes as well as ``SkyCoord`` be able to
+store and handle data has resulted in a large amount of code
+duplication (not just in the code proper, but also in the tests,
+leading to missed combinations). Restructuring the ``Frame``
+classes to remove data storage will allow for much more
+maintainable, de-duplicated code. It will also make it easier to
+contribute: if there is a problem one would like to solve in a
+given method, if one looks in ``SkyCoord``, one will likely find
+that it does not exist, but instead is defined on
+``BaseCoordinateFrame`` and gets dynamically called via
+``SkyCoord.__getattr__``. Indeed, the construction of
+``BaseCoordinateFrame`` ends up complicating ``SkyCoord``, which
+has to manage the coordinate data **through** the stored
+``BaseCoordinateFrame``.
+
+Another issue with the current implementation of coordinate
+frames is that the optional inclusion of coordinate data makes
+the reference frames “multi-modal”. This creates different usage
+modes (with and without data), each exhibiting different
+behavior. For instance, some methods such as ``separation`` work
+fine with frames with data, while doing this on coordinate frames
+without data results in an error. While this multi-modal
+structure as motivated in `APE 5
+<https://github.com/astropy/astropy-APEs/blob/main/APE5.rst>`_
+can be seen as a benefit for interpretability of the logic, we
+also now see it from the perspective of an anti-pattern. It
+forces any code interacting with the frame classes to handle both
+cases (checking ``frame.has_data``), complicating the codebase.
+Moreover static analyzers cannot determine which case is being
+used; with separated frames and data static analyzers will be
+able to prevent this entire class of errors. We thus find it
+worthwhile to separate coordinate data from reference frames at
+the possible expense of developers having to learn this new
+framework.
+
+All the points discussed thus far – separation of concerns and
+code duplication – concern maintainers. However user experience
+is the more important consideration. In this arena too,
+separating frames from data storage has its advantages. Perhaps
+most importantly, documentation will be more obvious: the methods
+and attributes are defined on ``SkyCoord`` (and ``Coordinate``)
+proper, so sphinx will know how to typeset those, while type
```

Comment:
Not very sure what this means but also maybe unnecessary to call out Sphinx specifically, as our code design should not be dictated by our choice of documentation renderer. But if you choose to keep the text, should capitalize and link, as follows.

```suggestion
proper, so `Sphinx <https://www.sphinx-doc.org/>`_ will know how to typeset those, while type
```